### ISSUE TITLE
Check frameworks behavior under inference mode

### ISSUE THREAD

**Issue by maximilianwerk** [jina-ai/finetuner#187]
Write a unit test, that checks, if batching vs. non-batching produce same results for a model that has `BatchNormlization` or `DropoutLayer` for: https://github.com/jina-ai/finetuner/blob/ae8e3990080681a760f465b29c381ffe0e4b0245/finetuner/embedding.py

**Comment by hanxiao** [comment]
https://github.com/jina-ai/finetuner/blob/main/tests/unit/test_embedding.py#L39-L74
